src/APL/vercel-kv/vercel-kv-apl.ts

```diff
@@ -29,44 +34,133 @@ export class VercelKvApl implements APL {
   async get(saleorApiUrl: string): Promise<AuthData | undefined> {
     this.debug("Will call Vercel KV to get auth data for %s", saleorApiUrl);
 
-    try {
-      const authData = await kv.hget<AuthData>(this.hashCollectionKey, saleorApiUrl);
-
-      return authData ?? undefined;
-    } catch (e) {
-      this.debug("Failed to get auth data from Vercel KV");
-      this.debug(e);
-
-      throw e;
-    }
+    return this.tracer.startActiveSpan(
+      "VercelKvApl.get",
+      {
+        attributes: {
+          saleorApiUrl,
+          [SemanticAttributes.PEER_SERVICE]: OTEL_APL_SERVICE_NAME,
+          [SemanticAttributes.HTTP_METHOD]: "GET",
+        },
+        kind: SpanKind.CLIENT,
+      },
+      async (span) => {
+        try {
+          const authData = await kv.hget<AuthData>(this.hashCollectionKey, saleorApiUrl);
+
+          this.debug("Received response from VercelKV");
+
+          if (!authData) {
+            this.debug("AuthData is empty for %s", saleorApiUrl);
+          }
+
+          span
+            .setStatus({
+              code: 200,
+              message: "Received response from VercelKV",
+            })
+            .end();
+
+          return authData ?? undefined;
+        } catch (e) {
+          this.debug("Failed to get auth data from Vercel KV");
+          this.debug(e);
+
+          span.recordException("Failed to get auth data from Vercel KV");
+
+          span
+            .setStatus({
+              code: SpanStatusCode.ERROR,
+              message: "Failed to get auth data from Vercel KV",
+            })
+            .end();
+
+          throw e;
+        }
+      }
+    );
   }
 
   async set(authData: AuthData): Promise<void> {
     this.debug("Will call Vercel KV to set auth data for %s", authData.saleorApiUrl);
 
-    try {
-      await kv.hset(this.hashCollectionKey, {
-        [authData.saleorApiUrl]: authData,
-      });
-    } catch (e) {
-      this.debug("Failed to set auth data in Vercel KV");
-      this.debug(e);
-
-      throw e;
-    }
+    return this.tracer.startActiveSpan(
+      "VercelKvApl.set",
+      {
+        attributes: {
+          saleorApiUrl: authData.saleorApiUrl,
```

Comment:
```suggestion
          saleorApiUrl: authData.saleorApiUrl,
          appId: authData.appId,
```

I think we should add `appId`, the same way as in Cloud APL